### PR TITLE
READY: RTS-311 / RTS-407 Timeseries support

### PR DESCRIPTION
**Pull request by lukebakken**
RTS-311, RTS-407

### CODE REVIEW

**Comment by lukebakken** [comment]
Depends on basho/riak_pb#135

File: lib/commands/ts/data.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+var ColumnType = Object.freeze({
+    Binary:    TsColumnType.BINARY,
+    Integer:   TsColumnType.INTEGER,
+    Numeric:   TsColumnType.NUMERIC,
+    Timestamp: TsColumnType.TIMESTAMP,
+    Boolean:   TsColumnType.BOOLEAN,
+    Set:       TsColumnType.SET,
+    Map:       TsColumnType.MAP,
+});
```
Comment (alexmoore) [inline lib/commands/ts/data.js]: Float + double?

File: lib/commands/ts/query.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            }
+        }
+    }
+
+    this._callback(null, response);
+    return true;
+};
+
+var schema = Joi.object().keys({
+    query: Joi.string().optional(), // TODO RTS-311 why is this optional?
```
Comment (alexmoore) [inline lib/commands/ts/query.js]: Good question.

File: lib/commands/ts/data.js (hunk 1 above)
Comment (lukebakken) [inline lib/commands/ts/data.js]: That would be RTS-426, which is still TODO. Is the support there Riak-side?

File: lib/commands/ts/storevalue.js [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        this.table = table;
+        return this;
+    },
+    /**
+     * Set the columns (optional).
+     * @method withColumns
+     * @param {Object[]} columns the timeseries columns in Riak
+     * @chainable
+     */
+    withColumns : function(columns) {
```
Comment (alexmoore) [inline lib/commands/ts/storevalue.js]: The Columns API won't be in 1.0 :/  Should we leave this in (and have it do nothing), or take it out for the release?

Comment (lukebakken) [inline lib/commands/ts/storevalue.js]: Using it didn't cause an error in my testing. I would prefer to leave this in.

File: test/unit/ts/query.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+               .build();
+            var protobuf = queryCommand.constructPbRequest();
+
+            var tsi = protobuf.getQuery();
+            assert(tsi, 'expected Tsinterpolation');
+
+            var base = tsi.getBase().toString('utf8');
+            assert.strictEqual(base, queryText);
+
+            var i = tsi.getInterpolations();
```
Comment (alexmoore) [inline test/unit/ts/query.js]: Same with interpolations, not in 1.0, but will be in 1.1...

Comment (lukebakken) [inline test/unit/ts/query.js]: It is only testing that it's an empty array in the `TsQueryReq` message.

File: test/integration/ts/timeseries.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+var TS = require('../../../lib/commands/ts');
+var RiakNode = require('../../../lib/core/riaknode');
+var RiakCluster = require('../../../lib/core/riakcluster');
+
+var assert = require('assert');
+var logger = require('winston');
+
+var tableName = 'GeoCheckin';
+var now = Date.now(); // NB: returns millis since Jan 1 1970 UTC
```
Comment (alexmoore) [inline test/integration/ts/timeseries.js]: So if we do this, and do multiple test runs, the tests that count result #'s will fail.  Use a static number, or perhaps something else?

**Comment by alexmoore** [comment]
Unsure of what happened with 21c0de848bfe194a8b45f2ba22923432428d1a6b, but nulls work now with my test change, and the latest riak_pb changes.

File: test/integration/ts/timeseries.js (hunk 5 above)
Comment (lukebakken) [inline test/integration/ts/timeseries.js]: OK a static number will work.

**Comment by alexmoore** [comment]
Merge master back in plz. Then :shipit:

**Pull request merged**